### ISSUE TITLE
Improvements for rule 'Local variables should not hide parent scope variables'

### ISSUE THREAD

**Issue by Ton-Pe** [Riverside-Software/sonar-openedge#1043]
Currently (OpenEdge rules version 2.20), if procedure has an input parameter defined and that same value is passed forward to an internal procedure as an input parameter with same name, this is flagged by SonarQube. Same goes for output parameters too. This is quite a typical usecase to pass value forward with same name to an internal procedure and causes many unnecessary issues reported for us. Rule has no supress issue annotation.

Example code
```
DEFINE INPUT  PARAMETER iItemCode AS CHARACTER NO-UNDO.
DEFINE OUTPUT PARAMETER oPrice    AS CHARACTER NO-UNDO.

RUN GetPrice(INPUT iItemCode, OUTPUT oPrice).

PROCEDURE GetPrice:

    DEFINE INPUT  PARAMETER iItemCode AS CHARACTER NO-UNDO. // these both are flagged by the rule
    DEFINE OUTPUT PARAMETER oPrice    AS CHARACTER NO-UNDO.

    /* Fetch price */
    FIND LAST Price WHERE Price.ItemCode = iItemCode NO-LOCK NO-ERROR.
    IF AVAILABLE Price THEN
        ASSIGN oPrice = Price.Price.

END PROCEDURE.
```

Could you please add at least a suppress annotation for this? Or optionally a parameter to the rule that would direct scanner to skip the usecases where the variables are input or output parameters?

**Comment by gquerret** [comment]
Merged in develop branch
